jmetal-core/src/main/java/org/uma/jmetal/util/experiment/component/EvolutionChart/HistoryData.java

```diff
@@ -0,0 +1,162 @@
+import org.uma.jmetal.qualityindicator.impl.Spread;
+import org.uma.jmetal.qualityindicator.impl.hypervolume.PISAHypervolume;
+import org.uma.jmetal.solution.DoubleSolution;
+import org.uma.jmetal.util.experiment.Experiment;
+import org.uma.jmetal.util.experiment.util.ExperimentAlgorithm;
+import org.uma.jmetal.util.front.Front;
+import org.uma.jmetal.util.front.imp.ArrayFront;
+import org.uma.jmetal.util.front.util.FrontNormalizer;
+import org.uma.jmetal.util.front.util.FrontUtils;
+import org.uma.jmetal.util.point.util.PointSolution;
+
+/**
+ *
+ * @author Lucas Prestes <lucas.prestes.lp@gmail.com> 
+ */
+public class HistoryData {
+    public static final String DEFAULT_BASE = "history";
+    private Integer testId;
+    
+    private Integer numberOfData;
+    private Integer numberOfTest;
+    
+    /**
+     * Number of data (%) / number of test
+     */
+    private Double[][] history;
+   
+
+    public HistoryData(Integer numberOfTest) {
+        this.testId = -1;
+        this.numberOfData = 100;
+        this.numberOfTest = numberOfTest;
+        history = new Double[this.numberOfData][this.numberOfTest];
+
+        for (int i = 0; i < this.numberOfData; i++) {
+            for (int j = 0; j < this.numberOfTest; j++) {
+                history[i][j] = 0.0;
+            }
+        }
+        
+    }
+    
+    public void addData(Double data, Integer timeIndex){
+        if(timeIndex==0)testId++;
+        history[timeIndex][testId] = data;
+    }
+    
+    /**
+     * Will save each line how an test
+     * @param baseDir will create the directory if doesn't exist
+     * @param indicator for file name
+     * @param nameAlgorithm for file name
+     * @param problemName  for file name
+     */
+    public void printAllHistory(String baseDir, String indicator, String nameAlgorithm, String problemName){
+        String outputDir = baseDir + "/"+DEFAULT_BASE+"/"+nameAlgorithm+"/";
+        File f = new File(outputDir);
+        if(!f.exists()){
+            f.mkdirs();
+        }
+        try (OutputStream os = new FileOutputStream(outputDir+"data_"+indicator+"_"+nameAlgorithm+"_"+problemName+".dat")) {
+            PrintStream ps = new PrintStream(os);
+            for (int data = 0; data < numberOfData; data++) {
+                ps.print(data);
+                if(data<numberOfData-1){
+                    ps.print("\t");
+                }
+            }
+            ps.print("\n");
+            for (int test = 0; test < numberOfTest; test++) {
+                for (int data = 0; data < numberOfData; data++) {
+                    ps.print(history[data][test]);
+                    if(data<numberOfData-1){
+                        ps.print("\t");
+                    }
+                }
+                ps.print("\n");
+            }
+        } catch (IOException ex) {
+            Logger.getLogger(GenerateEvolutionChart.class.getName()).log(Level.SEVERE, null, ex);
```

Comment:
I would say wrong class in the logger. And I would suggest to declare the logger as a static member of the class, so we can reuse it anywhere in the class.